### PR TITLE
log: Change EnabledParameters to have a field instead of getter and setter

### PR DESCRIPTION
**Pull request by pellared**
## What

Allow users to to write less verbose code by changing methods to field so that the callers can do:

```go
if logger.Enabled(ctx, log.EnabledParameters{ Severity: log.SeverityDebug }) {
 // ...
}
```

instead of

```go
params := log.EnabledParameters{}
params.SetSeverity(log.SeverityDebug)
if logger.Enabled(ctx, params) {
 // ...
}
```

## Why

- We may provide a `Logger.EnabledEvent` user-facing API in close future and it would be nice if the signatures are similar. See: https://github.com/open-telemetry/opentelemetry-specification/issues/4263. The possible alternative is that user-facing `Logger.Enabled` would also be used for events as well.

- There is a chance that the Logger API is going to be user-facing. See: https://github.com/open-telemetry/opentelemetry-specification/issues/4234.

- I (subjectively) believe that ~~having a shorter type name and~~ using fields instead of methods has a better user experience.
Ideally the user would prefer having `Enabled(ctx context.Context, severity SeverityLevel) bool`.
However, we would be not able to provide additional parameters without creating new methods.

- I was thinking for for a few weeks before proposing this PR. I was worried if fields would be able to handle all possible cases. However, we can always add methods for future parameters. Having mixed fields and methods should not be the end of the world - especially that it is e.g. how [`slog.Record`](https://pkg.go.dev/log/slog#Record) is designed. We were discussing inside Logs SIG if we ever would like to add (partial) attributes or (partial) body to `Logger.Enabled`. We all agreed that we do not want it as it can introduce overhead that we want to avoid and also it can lead to a very confusing API experience. Moreover, event name is being promoted to a top-level field of a log record - this was the only "attribute" that seem to be important, but there are even more reasons to promote it to a top-level field. See: https://github.com/open-telemetry/opentelemetry-specification/issues/4260. Therefore, there is a pretty low chance that we will introduce any complex parameter in future.

### CODE REVIEW

**Comment by dashpole** [comment]
Do we have any other examples of FooParameters or FooParams?  Consistency across the codebase is important to me

**Comment by MrAlias** [comment]
https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#SamplingParameters

**Comment by pellared** [comment]
> Do we have any other examples of FooParameters or FooParams? Consistency across the codebase is important to me

Renaming back so this PR would only propose to have fields instead of getters and setters.

**Comment by pellared** [comment]
@dashpole, @MrAlias, renamed back to `EnabledParameters`.

There is need to rush with this PR. We can discuss it during SIG meeting.

**Review by dashpole — APPROVED** [review]

**Comment by dashpole** [comment]
I'm definitely ok with this.  Having a struct for sampling has worked well so far, and this seems similar.

File: log/logtest/recorder_test.go [hunk 1]
```diff
@@ -68,17 +68,17 @@ func TestRecorderLoggerCreatesNewStruct(t *testing.T) {
...
 func TestLoggerEnabled(t *testing.T) {
 	for _, tt := range []struct {
-		name                   string
-		options                []Option
-		ctx                    context.Context
-		buildEnabledParameters func() log.EnabledParameters
+		name               string
+		options            []Option
+		ctx                context.Context
+		buildEnabledParams func() log.EnabledParameters
```
Comment (MrAlias) [inline log/logtest/recorder_test.go:74]: nit: no need for a function here anymore.

```suggestion
		buildEnabledParams log.EnabledParameters
```

**Review by MrAlias — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

File: log/logtest/recorder_test.go (hunk 1 above)
Comment (pellared) [reply log/logtest/recorder_test.go:74]: Nice catch! I will address it before merging.

Comment (pellared) [reply log/logtest/recorder_test.go]: PTAL https://github.com/open-telemetry/opentelemetry-go/pull/6009/commits/a3f93f168432d9d9ee6dabc210e7f3d79dfd06b7

**Pull request merged**